src/pages/workspaces/workspace/notebooks/TerminalLauncher.js

```diff
@@ -64,7 +64,8 @@ const TerminalLauncher = _.flow(
   }
 
   async componentDidUpdate() {
-    const { cluster: { clusterUrl } = {} } = this.props
+    const { cluster } = this.props || {}
+    const { clusterUrl } = cluster || {}
```

Comment:
`clusterUrl` is expected to be a string, but if `cluster` is falsy then it'll be an empty object. The way it was before, if `cluster` was falsy, `clusterUrl` would be `undefined`. This changes the truthiness relationship between `cluster` and `clusterUrl`. Is that intended?

Reply:
Thanks for pointing that out, I misinterpreted how the de-structure worked in the original version. Only cluster should be set to `{}` if it doesn't exist, I will remove that `|| {}` from line 68.